internal/storage/image.go

```diff
@@ -638,7 +638,12 @@ func pullImageChild() {
 		os.Exit(1)
 	}
 
-	output <- pullImageOutputItem{Result: transports.ImageName(destRef), Name: canonicalRef.Name(), Digest: canonicalRef.Digest()}
+	rawCanonical, ok := canonicalRef.Raw().(reference.Canonical)
+	if !ok {
+		fmt.Fprintf(os.Stderr, "Returned reference %v is not canonical", canonicalRef.Raw().String())
```

Comment:
can you use the `cri-o/internal/log` package here instead of logging right to Stderr?

Reply:
I see this (and my below comment) are done above, could you cleanup both cases?

Reply:
Both of these are in `pullImageChild`, a separate process. The error message is explicitly expected to go to Stderr, and `pullImageParent` reads that (as `errOutput`) and turns it into a Go error.

From skimming `internal/log`, it seems to add Logrus’ typical log metadata; I think that should not be included in the Go error returned from `pullImageParent`.